hook/consul/hook.go

```diff
@@ -46,15 +46,15 @@ type Config struct {
 
 // HandleEvent calls appropriate hook functions that correspond to supported
 // event types. Unsupported events are ignored.
-func (h *Hook) HandleEvent(event hook.Event) error {
+func (h *Hook) HandleEvent(event hook.Event) (hook.Env, error) {
 	switch event.Type {
 	case hook.AfterTaskHealthyEvent:
-		return h.RegisterIntoConsul(event.TaskInfo)
+		return hook.Env{}, h.RegisterIntoConsul(event.TaskInfo)
 	case hook.BeforeTerminateEvent:
-		return h.DeregisterFromConsul(event.TaskInfo)
+		return hook.Env{}, h.DeregisterFromConsul(event.TaskInfo)
 	default:
 		log.Debugf("Received unsupported event type %s - ignoring", event.Type)
-		return nil // ignore unsupported events
+		return hook.Env{}, nil // ignore unsupported events
```

Comment:
IMO `nil` will be better than `hook.Env{}`

Reply:
It's not a problem really - you have to perform error check before you will use that value.

Reply:
And deal with mocking in tests. Will try that approach (again).

Reply:
I mean, it does not make sense to return pointer here.

Reply:
Returning nil then.